prometheus/base/prometheus.rules.yaml

```diff
@@ -15,3 +15,22 @@ spec:
             * on (namespace, pod)
             group_left (label_pipelines_appstudio_openshift_io_type)
             kube_pod_labels
+        # containers and init containers resource limits with added pod labels
+        - record: appstudio_container_resource_limits_with_additional_labels
+          expr: |
+            {__name__=~"kube_pod_.*container_resource_limits"}
+            * on (namespace, pod)
+            group_left(label_pipelines_appstudio_openshift_io_type) kube_pod_labels
+        - record: labeled_containers:resource_minutes:2w
+          expr: |
+            last_over_time(appstudio_container_resource_limits_with_additional_labels[2w])
+            * on (namespace, pod, container) group_left ()
+            (sum by (namespace, pod, container) (increase(container_last_seen[2w] )) / 60)
+    - name: resource_gauge
+      interval: 5m
+      rules:
+        - record: appstudio_container_resource_minutes_gauge
```

Comment:
Looks pretty good to me :+1:

Reply:
@hmariset @Omeramsc the rule defined here can be used for resource minutes for both cpu and memory in both pipelines and deployments (might require additional labels for differentiation between the different uses).

I'm off today (not sure about tomorrow). It can be good if you can start coming up with unit tests for the rule - we can do that on a couple of PR and eventually consolidate them and merge only one of them.